x-pack/plugins/apm/public/components/shared/redirect_with_offset/index.tsx

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useLocation, Redirect } from 'react-router-dom';
+import qs from 'query-string';
+import React from 'react';
+import moment from 'moment';
+import { useApmRouter } from '../../../hooks/use_apm_router';
+import { getDateRange } from '../../../context/url_params_context/helpers';
+
+import {
+  TimeRangeComparisonEnum,
+  getSelectOptions,
+} from '../../../components/shared/time_comparison/get_comparison_options';
+
+export function RedirectWithOffset({
+  children,
+}: {
+  children: React.ReactElement;
+}) {
+  const location = useLocation();
+  const query = qs.parse(location.search);
+
+  const apmRouter = useApmRouter();
+  const matchingRoutes = apmRouter.getRoutesToMatch(location.pathname);
+  const matchesRoute = matchingRoutes.some((route) => {
+    return (
+      route.path === '/services' ||
+      route.path === '/traces' ||
+      route.path === '/service-map' ||
+      route.path === '/backends' ||
+      route.path === '/services/{serviceName}' ||
+      route.path === '/service-groups' ||
+      location.pathname === '/' ||
+      location.pathname === ''
+    );
+  });
+
+  if (
+    'comparisonType' in query &&
+    'rangeFrom' in query &&
+    'rangeTo' in query &&
+    matchesRoute &&
+    Object.values(TimeRangeComparisonEnum).includes(
+      query.comparisonType as TimeRangeComparisonEnum
+    )
+  ) {
+    const { rangeFrom, rangeTo, comparisonType, ...queryRest } = query;
+
+    const { start, end } = getDateRange({
+      rangeFrom: rangeFrom as string,
+      rangeTo: rangeTo as string,
+    });
+
+    const momentStart = moment(start);
```

Comment:
Instead of getting `rangeFrom` / `rangeTo` and calculating the diff I can just map `TimeRangeComparisonEnum.DayBefore` to `1d` and `TimeRangeComparisonEnum.WeekBefore` to `1w`. Then the value is `PeriodBefore` we don't have any other options which means even without redirects we will still get the default value. @dgieselaar WDYT?

Reply:
sgtm, although I'm not sure how that helps with redirect - if users have bookmarks or whatever it will still include the literal string no? am I missing something?

Reply:
It would still calculate the right offset because it'll be the default option, but `comparisonType=period` will remain in the url. I only mentioned it as it'll simplified the redirect a bit. I'm find to leave it as is if you think it looks ok.

Reply:
oh I see now I think, I assumed you were talking about changing the value of the enum. You're talking about mapping it here? would that also remove the need to call `getSelectOptions` below?

Reply:
Yes, it would be removed. It'll be like `if (comparisonType === 'day') offset = '1d')` (same for the week). For period we just leave it continue and the time comparison component will calculate the default.

Reply:
sgtm 👍

Reply:
Done in https://github.com/elastic/kibana/pull/127812/commits/de4d405d3174f947515c981be4e39b061b2573dc